app/views/curation_concerns/base/_form_progress.html.erb

```diff
@@ -22,7 +22,7 @@
     <div class="panel-footer text-center">
       <% if Sufia::Engine.config.active_deposit_agreement_acceptance %>
           <label>
-            <%= check_box_tag 'agreement', 1, nil, required: true %>
+            <%= check_box_tag 'agreement', 1, defined?(agreement_accepted) && agreement_accepted, required: true %>
```

Comment:
The `defined?(agreement_accepted)` check is needed because the `_form_progress` is rendered from more than one place. Instead of this check I could pass the `agreement_accepted` from all places where the form is rendered.